### PR TITLE
Configuration & Route lifecycle operations are required

### PR DESCRIPTION
**Pull request by dprotaso**
This allows for portability of these resources for advanced users
who do not use Knative Service

### CODE REVIEW

**Comment by eallred-google** [comment]
I don't think we should change this too required.  Can you provide more rationale regarding making this change?  Portability could arguably be applied to every API setting and call.

**Comment by salaboy** [comment]
I do agree with @dprotaso in the sense that Knative clients like the CLI tools are already using those APIs.

**Comment by dprotaso** [comment]
> Can you provide more rationale regarding making this change? 

Configuration & Route were never marked internal so integrators always had the option to drop to these APIs for flexibility. There's already an example of this where [projectriff](https://projectriff.io) (now archived) was using configuration and route directly. I believe this was done to get around some issues with revision management.

A use case I'm aware of is the need for stable routing over configurations instead of revisions from a single Knative Service.

File: specs/serving/knative-api-specification-1.0.md [hunk 1]
```diff
@@ -362,12 +362,6 @@ code and configuration states (Revisions).
 Routes which are owned (controlled) by a Service SHOULD NOT be updated by
 developers; any changes will be reset by the Service controller.
 
-The table below details which operations must be made available to a developer
-accessing a Knative Route using a minimal profile. For any non-minimal profile,
-the POST, PUT, or DELETE operations MUST be enabled as a group. This ensures
-that the developer has the ability to control the complete lifecycle of the
-object from create through deletion.
```
Comment (evankanderson) [inline specs/serving/knative-api-specification-1.0.md:369]: Why remove this verbiage?

**Review by evankanderson — COMMENTED** [review]
I think it's okay for some clients (and maybe even the OSS CLI) to use operations which aren't marked as REQUIRED in the spec.

For a specific example, I think this may break Cloud Run (Fully Managed), because I'm not sure that they have support for free-range Configurations without an associated Service. @whaught to confirm here, since I honestly don't remember.

**Comment by whaught** [comment]
> I think it's okay for some clients (and maybe even the OSS CLI) to use operations which aren't marked as REQUIRED in the spec.
> 
> For a specific example, I think this may break Cloud Run (Fully Managed), because I'm not sure that they have support for free-range Configurations without an associated Service. @whaught to confirm here, since I honestly don't remember.

This would likely break Cloud Run. Internally we do support actions on Configurations and Routes and it's likely they could be made public, but doing so is out of scope for the simple experience we are intending to provide. Ideally this would not be required.

File: specs/serving/knative-api-specification-1.0.md (hunk 1 above)
Comment (dprotaso) [reply specs/serving/knative-api-specification-1.0.md:369]: POST, PUT, or DELETE operations were marked as REQUIRED in tandem. The only remaining OPTIONAL operation is PATCH so it didn't seem necessary to keep this clause

**Comment by dprotaso** [comment]
>  but doing so is out of scope for the simple experience we are intending to provide. Ideally this would not be required.

This would offer integrators who might need to drop to lower levels portability.

**Comment by yanweiguo** [comment]
From #412, I think the de-composed resource model wasn't compatible with our ability to deliver a great developer experience so we introduced `Service`. IMO we should encourage developers to use service directly. Marking Configuration & Route create/update API as REQUIRED sounds an opposite to simple experience to me.

**Comment by dprotaso** [comment]
> From knative/serving#412, I think the de-composed resource model wasn't compatible with our ability to deliver a great developer experience so we introduced Service

I agree that `Service` is the abstraction that developers should use

> Marking Configuration & Route create/update API as REQUIRED sounds an opposite to simple experience to me.

I'm suggesting these resources are for 'integrators' of Knative who understand the extra complexity in exchange for flexibility

**Comment by xtreme-sameer-vohra** [comment]
[Serving overview](https://github.com/knative/specs/blob/main/specs/serving/overview.md#service) has an interesting description thats relevant here
```
Service acts only as an orchestrator of the underlying Route and Configuration (much as a kubernetes Deployment orchestrates ReplicaSets). Its usage is optional but recommended.
```

**Comment by whaught** [comment]
Right. Integrators should be fine to use Route and Configuration directly in their implementation of the spec if they choose. However we also think offerings like Cloud Run which aim to offer a simplified experience should also be allowed to lock down the experience to one simple endpoint (Service) without it being required that we allow further complexity from clients. Either should still be considered in compliance with Knative which allows for diversity in the experiences vendors wish to offer.

This difference ought not be _required_ of any vendor lest a particular implementation become the de-facto specification.

**Comment by dprotaso** [comment]
This was veto'd by Google (@spencerdillard)

**Pull request closed without merging**